rest/kadai-rest-spring/src/test/java/io/kadai/user/rest/UserControllerIntTest.java

```diff
@@ -200,9 +205,274 @@ void should_ReturnExistingUsers_When_ParameterContainsDuplicateAndInvalidIds() {
+              } else if (level == 2) {
+                assertThat(user.getOrgLevel2()).isEqualTo(orgLevel);
+              } else if (level == 3) {
+                assertThat(user.getOrgLevel3()).isEqualTo(orgLevel);
+              } else if (level == 4) {
+                assertThat(user.getOrgLevel4()).isEqualTo(orgLevel);
+              }
+            });
+  }
+
+  @ParameterizedTest
+  @CsvSource({"KADAI,1", "Human Workflow,2", "BPM,3", "Envite,4"})
+  void should_ReturnIntersection_When_OrgLevelAndCurrentUserAreGiven(
+      String orgLevel, int level) {
+    String url =
+        restHelper.toUrl(RestEndpoints.URL_USERS)
+            + String.format("?orgLevel%d=%s", level, orgLevel)
+            + "&current-user";
+    HttpEntity<?> auth = new HttpEntity<>(RestHelper.generateHeadersForUser("teamlead-1"));
+
+    ResponseEntity<UserCollectionRepresentationModel> responseEntity =
+        TEMPLATE.exchange(
+            url,
+            HttpMethod.GET,
+            auth,
+            ParameterizedTypeReference.forType(UserCollectionRepresentationModel.class));
+
+    assertThat(responseEntity.getBody()).isNotNull();
+    assertThat(responseEntity.getBody().getContent())
+        .haveExactly(
+            1,
+            new Condition<>(
+                user -> user.getUserId().equals("teamlead-1"),
+                "user with id teamlead-1"));
+    responseEntity.getBody().getContent().stream()
+        .filter(user -> !user.getUserId().equals("teamlead-1"))
+        .forEach(
+            user -> {
+              if (level == 1) {
+                assertThat(user.getOrgLevel1()).isEqualTo(orgLevel);
+              } else if (level == 2) {
+                assertThat(user.getOrgLevel2()).isEqualTo(orgLevel);
+              } else if (level == 3) {
+                assertThat(user.getOrgLevel3()).isEqualTo(orgLevel);
+              } else if (level == 4) {
+                assertThat(user.getOrgLevel4()).isEqualTo(orgLevel);
+              }
+            });
+  }
+
+  @ParameterizedTest
+  @CsvSource({"KADAI,1", "Human Workflow,2", "BPM,3", "Envite,4"})
+  void should_ReturnIntersection_When_OrgLevelAndUserIdsAndCurrentUserAreGiven(
+      String orgLevel, int level) {
+    String url =
+        restHelper.toUrl(RestEndpoints.URL_USERS)
+            + String.format("?orgLevel%d=%s", level, orgLevel)
+            + "&current-user"
+            + "&user-id=user-1-1"
+            + "&user-id=user-2-1"
+            + "&user-id=teamlead-1";
+    HttpEntity<?> auth = new HttpEntity<>(RestHelper.generateHeadersForUser("teamlead-1"));
+
+    ResponseEntity<UserCollectionRepresentationModel> responseEntity =
+        TEMPLATE.exchange(
+            url,
+            HttpMethod.GET,
+            auth,
+            ParameterizedTypeReference.forType(UserCollectionRepresentationModel.class));
+
+    assertThat(responseEntity.getBody()).isNotNull();
+    assertThat(responseEntity.getBody().getContent())
```

Comment:
how about defining    

` List<User> users = userCollection.getContent(); `

since we use it multiple times in the following asserts?

and then refer like to it like e.g.:

```
  assertThat(users)
            .filteredOn(user -> "teamlead-1".equals(user.getUserId()))
            .hasSize(1);
```
            
 This is probably just flavour, not really obligatory. What do you think?